cms/djangoapps/contentstore/views/item.py

```diff
@@ -103,6 +103,16 @@ def _filter_entrance_exam_grader(graders):
     return graders
 
 
+def _is_library_component_limit_reached(usage_key):
+    """
+    Verify if the library has reached the maximum number of components allowed in it
+    """
+    store = modulestore()
+    parent = store.get_item(usage_key)
+    total_children = len(parent.get_children())
```

Comment:
Will this load one block (the parent), or the parent plus all of its children?

If the latter, do you think that's worth optimizing?

I'm not especially familiar with modulestore performance considerations, so I'm open to deferring to your judgement here.

Reply:
Good catch - have optimized this to not load all the children, which can be quite a lot. Thanks!